lib/event.js

```diff
@@ -15,8 +15,64 @@ class EventModel extends BaseModel {
         super('event', config);
     }
 
+    /**
+     * Return stage builds that belong to this event
+     * @method getStageBuilds
+     * @return {Promise}  Resolves to an array of stage builds
+     */
+    async getStageBuilds() {
+        const nodes = this.workflowGraph.nodes;
+        const stageNamesSet = new Set();
+
+        nodes.forEach(n => {
+            if (n.stageName) {
+                stageNamesSet.add(n.stageName);
+            }
+        });
+
+        const stageNames = Array.from(stageNamesSet);
+        const stageListConfig = {
+            params: {
+                pipelineId: this.pipelineId,
+                name: stageNames,
+                archived: false
```

Comment:
Entire logic to get the `stages` and merge some fields (`name`) into `stageBuilds` need not be done here.

If we decide to fetch `stages`, we should get the `stages` even if it is archived. If there is a reference to the archived stage in the workflow graph, means the stage was active when the event got created.
```suggestion
                name: stageNames
```

Reply:
removed